test/functional/validation-options.spec.ts

```diff
@@ -1086,4 +1086,30 @@ describe('context', () => {
       expect(errors[1].contexts['contains']).toEqual({ bye: 'now' });
     });
   });
+
+  it('should stop at first error.', () => {
+    class MyClass {
+      @IsDefined({
+        message: 'isDefined',
+        context: {
+          bye: 'now',
+        },
+      })
+      @Contains('hello', {
+        message: 'String is not valid. You string must contain a hello word',
+        context: {
+          bye: 'now',
+        },
+      })
+      sameProperty: string;
+    }
+
+    const model = new MyClass();
+    return validator.validate(model, { stopAtFirstError: true }).then(errors => {
+      expect(errors.length).toEqual(1);
+      expect(Object.keys(errors[0].constraints).length).toBe(1);
+      expect(errors[0].contexts['isDefined']).toEqual({ bye: 'now' });
+      expect(errors[0].constraints['isDefined']).toBe('isDefined');
+    });
+  });
```

Comment:
Also, add a test here when `stopAtFirstError` is set to true. (I know this is the default behavior and have a lot of implicit test for this, but let's test it explicitly too.

Reply:
I am writing here to test when `stopAtFirst` is set to true.
Are you saying that we should also test for false in this test, as shown below?

```ts
return validator.validate(model, { stopAtFirstError: true }).then(errors => {
      expect(errors.length).toEqual(1);
      expect(Object.keys(errors[0].constraints).length).toBe(1);
      expect(errors[0].contexts['isDefined']).toEqual({ bye: 'now' });
      expect(errors[0].constraints['isDefined']).toBe('isDefined');
    }).then(() => validator.validate(model, { stopAtFirstError: false })).then(errors => {
      expect(Object.keys(errors[0].constraints).length).toBe(2);
    });
```

Reply:
@NoNameProvided

Reply:
Yes, usually it's good to explicitly test the opposite to show the intent. (We have a lot of implicit test for the default behavior, but it's always better to be explicit. For every feature behind flag there should be a test for on and off version.)

I will go ahead and merge it as it is, in future contribution please keep this in mind. Thanks!

Reply:
Yes, sir.